compiler/damlc/tests/src/DA/Test/DamlRenamer.hs

```diff
@@ -0,0 +1,94 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE CPP #-}
+
+module DA.Test.DamlRenamer (main) where
+
+import Control.Monad (filterM)
+import DA.Bazel.Runfiles (exe, locateRunfiles, mainWorkspace)
+import Data.List.Extra (nubOrd)
+import Data.Text (Text)
+import System.Directory (doesFileExist, listDirectory, makeAbsolute)
+import System.Environment.Blank (setEnv)
+import System.Exit (ExitCode (..))
+import System.FilePath (dropExtension, replaceExtensions, takeExtensions, takeFileName, (<.>), (</>))
+import System.Process (readProcessWithExitCode)
+import Test.Tasty.Golden (goldenVsStringDiff)
+
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as TE
+import qualified Test.Tasty.Extended as Tasty
+
+main :: IO ()
+main = do
+  setEnv "TASTY_NUM_THREADS" "1" True
+  damlc <- locateRunfiles $ mainWorkspace </> "compiler" </> "damlc" </> exe "damlc"
+  testDir <- locateRunfiles $ mainWorkspace </> "compiler/damlc/tests/daml-test-files"
+  print (damlc, testDir)
+  Tasty.deterministicMain =<< allTests damlc testDir
+
+allTests :: FilePath -> FilePath -> IO Tasty.TestTree
+allTests damlc testDir = Tasty.testGroup "All Daml GHC tests using Tasty" <$> sequence
+  [ mkTestTree damlc testDir
+  ]
+
+mkTestTree :: FilePath -> FilePath -> IO Tasty.TestTree
```

Comment:
this is very similar to the `mkTestTree` functions from `DA.Daml.Doc.Tests`, `DA.Daml.Doc.Render.Tests` and `DA.Daml.Desugar.Tests`. In a later PR I'd like to factor out the common part into a new module at `libs-haskell/da-hs-base/src/Test/Tasty/Golden/Extended.hs`